### ISSUE TITLE
Change Request: Remove all formatters except stylish, html, json, and json-with-meta

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#17524]
### ESLint version

HEAD

### What problem do you want to solve?

We currently ship 12 formatters in the core of ESLint. This was helpful early on to get people started with ESLint and help them transition from existing tools. However, the results of the user survey show very few people use anything but the default formatter (stylish). 

Additionally, the `tap` formatter requires the `js-yaml` package, which isn't need anywhere else in the core, so removing this would mean eliminating a dependency. (Eventually -- `@eslint/eslintrc` also depends on `js-yaml`, but that dependency will be removed in v10.)

### What do you think is the correct solution?

Remove all of the formatters except stylish, json, and json-with-meta. Almost all of the remaining formatters have already been published as their own packages: https://github.com/fregante/eslint-formatters. We can advise people to use those packages if they need to continue using one of these formatters.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

Maybe we'd also like to keep `html`? It looks like `eslint-formatter-html` already exists and isn't the code that we have.

**Comment by JoshuaKGoldberg** [comment]
#17531 is ready for review. Marked as draft for now as it's a breaking change.

**Comment by mdjermanovic** [comment]
I'd definitely support removing formatters that are used to pass the results to other systems on CI, as they could be installed as project's dev-dependencies. That would be, I believe: `checkstyle`,  `jslint-xml`, `junit`, `tap`, `unix`(?).

Less sure about removing formatters that might be users' personal preferences: `compact`, `html`, `visualstudio`.

**Comment by nzakas** [comment]
We agreed in the TSC meeting to move forward with this and see what kind of feedback we receive.

**Comment by nzakas** [comment]
I think we'll need to plan on keeping `html`, as there is already an `eslint-config-html` package that is very different.

**Comment by MichaelDeBoey** [comment]
@nzakas This would break the packages defined in https://github.com/fregante/eslint-formatters as that repo has a script to just copy over the code from this repo

> I think we'll need to plan on keeping `html`, as there is already an `eslint-config-html` package that is very different.

We could publish all these formatters under the @eslint-community org if you want to, so we have a consistent place for _all_ of them

**Comment by fregante** [comment]
@nzakas  Totally, I've already done that for a couple other formatters that you decided to maintain in that organization, like `table`. 

So:

1. I'll add you (or others) as an owner to the desired npm names
2. You can set up your own repositories for each formatter
3. I can be removed as the owner whenever that's done
4. I'll remove the published packages from my `eslint-formatter` repo

Since eslint is still keeping some formatters in the `eslint` package itself, the `eslint-formatter` publisher will continue to work as usual, but for fewer formatters. 

As for the `html` formatter, the owner contacted me on GitHub to propose the transfer, but then never proceeded with that, even after a direct email.

**Comment by nzakas** [comment]
@fregante thanks! If you can add eslint-community-bot as an owner, that should do the trick. We can then use our existing credentials on the org to publish changes.

@MichaelDeBoey I'll leave it up to you how you'd like to set this up on the org. I'd suggest doing a monorepo to make maintenance easier.

**Comment by fregante** [comment]
Done
```
❯ npm owner add eslint-community-bot eslint-formatter-checkstyle
❯ npm owner add eslint-community-bot eslint-formatter-compact
❯ npm owner add eslint-community-bot eslint-formatter-jslint-xml
❯ npm owner add eslint-community-bot eslint-formatter-junit
❯ npm owner add eslint-community-bot eslint-formatter-tap
❯ npm owner add eslint-community-bot eslint-formatter-visualstudio
```

But I hope you keep `unix` in eslint, it's a simple formatter that uses clickable line:col references for each error. The grouped style of `stylish` has terrible UX when used in the console of VS Code, for example

**Comment by nzakas** [comment]
`unix` is gone too. I don't think it's too much to ask for folks to install the formatter they want.

**Comment by nzakas** [comment]
Just the ones mentioned here: https://github.com/eslint/eslint/pull/17531

I'd like to stick with `eslint-formatter-*` because it will allow people to avoid changing their existing CLI commands after installing the package (so `npx eslint . -f checkstyle` will continue to work once `eslint-formatter-checkstyle` is installed)

**Comment by sam3k** [comment]
2023-11-30 tsc-meeting note: waiting on the last remaining changes.

**Comment by fregante** [comment]
Where do these repos live now? I only saw the original `table` and `codeframe` on the ESLint Community organization. I added the bot as an owner last year but the packages still point to my repo: https://www.npmjs.com/~eslint-community-bot
